extension/flow/HttpFlow.js

```diff
@@ -84,96 +90,118 @@ class HttpFlow {
     }
   }
 
+  async getUserAgentInfo(userAgent) {
+    let result = uaInfoCache.peek(userAgent);
+    if (!result) {
+      result = {};
+      try {
+        const detectResult = this.detector.detect(userAgent);
+        if (detectResult)
+          result.detect = detectResult;
+        const parseResult = useragent.parse(userAgent);
+        if (parseResult)
+          result.parse = parseResult;
+        if (Object.keys(result) > 0)
+          uaInfoCache.set(userAgent, result);
```

Comment:
saving result is not necessary here. If match on LRU, just reset expire on redis key should be fine.